mlir/lib/Dialect/MIOpen/utility/transformMapUtils.cpp

```diff
@@ -410,19 +409,30 @@ propagateVectorizationInfo(TransformMapAttr map,
       SmallVector<std::tuple<int64_t, uint32_t>> data(zip.begin(), zip.end());
       std::sort(data.begin(), data.end());
 
+      // Note: with negative coefficients, we can't reliably vectorize, as
+      // the subtraction can push us off the left edge and we don't necessarily
+      // have an upper bound on how much will be subtracted. So, we use
+      // a length of 1 in the fastest vectorizing dimension we can find and
+      // call it a day.
+      bool hasNegativeCoefficients = false;
       Optional<VectorizationInfo> ourResult;
       for (auto pair : data) {
         int64_t coefficient = std::get<0>(pair);
         uint32_t upperDim = std::get<1>(pair);
+        if (coefficient < 0)
+          hasNegativeCoefficients = true;
         if (input[upperDim].hasValue()) {
           if (coefficient == 0)
             continue;
 
           int64_t upperLen = input[upperDim]->maxLength;
           int64_t needsCoeff = input[upperDim]->needsCoefficient;
 
-          if (!ourResult.hasValue())
+          if (!ourResult.hasValue()) {
             ourResult = VectorizationInfo(1, coefficient);
+            if (hasNegativeCoefficients)
+              break;
```

Comment:
We do need the flag, because this test is only being run when the current dimmension has vectorizatwion info attached. The negative coefficient might be on a dimesnion that's being assumed to be held constant.